visualize_data.ipynb

```diff
@@ -0,0 +1,264 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib\n",
+    "from matplotlib import pylab\n",
+    "%config InlineBackend.figure_format = 'retina'\n",
+    "matplotlib.rcParams['figure.dpi'] = 80\n",
+    "textsize = 'x-large'\n",
+    "params = {'legend.fontsize': 'large',\n",
+    "          'figure.figsize': (5, 4),\n",
+    "         'axes.labelsize': textsize,\n",
+    "         'axes.titlesize': textsize,\n",
+    "         'xtick.labelsize': textsize,\n",
+    "         'ytick.labelsize': textsize}\n",
+    "pylab.rcParams.update(params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can read in our training data from a `.npy` file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_fn = 'data/cosmology_train.npy'\n",
+    "train = np.load(train_fn, allow_pickle=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The file is saved as an array of dictionaries that have x (input parameters), y (output statistics), and labels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of datapoints: 1000\n",
+      "Dict keys: dict_keys(['x', 'y', 'label'])\n"
+     ]
+    }
+   ],
+   "source": [
+    "ntrain = train.shape[0]\n",
+    "print(\"Number of datapoints:\", ntrain)\n",
+    "print(\"Dict keys:\", train[0].keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The cosmology dataset contains cosmological parameters (Omega_m, sigma8, Omega_b) as inputs, and the correlation function as output. The correlation function is measured at 10 separation values $r$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of input parameters: 3\n",
+      "Number of output values: 10\n"
+     ]
+    }
+   ],
+   "source": [
+    "rvals = np.linspace(50, 140, 10)\n",
```

Comment:
Sometimes the variable names use `_` (snake case) to separate the words. Other times they don't. I think it would be better if all were consistent.

Reply:
This happens in some places but as the notebook has the results, I can not find it there (too much binary on the images)

Reply:
yes let's make sure theyre consistent! do you have a preference for snake case vs not?

Reply:
I like more `snake_case` because it is easier to read.